Sample/fhirdate.py

```diff
@@ -1,79 +1,287 @@
-#!/usr/bin/env python
-# -*- coding: utf-8 -*-
-#
-#  Facilitate working with dates.
-#  2014, SMART Health IT.
-
-import sys
-import logging
-import isodate
+"""Facilitate working with FHIR dates and times."""
+# 2014-2024, SMART Health IT.
+
 import datetime
+import re
+from typing import Any, Union
+
 
+class _FHIRDateTimeMixin:
```

Comment:
In fact, there's probably an argument that each of these classes should go in their own file (`fhirdatetime.py`, `fhirinstant.py` etc) just like each model. But because I wanted to share this code, I left them all here. But I'm open to splitting the files.

Reply:
i do :sort of: like the ability to say 'is this object a time related object', but I buy the mixin version.

I do think it might help with readability to have each class in separate files, or to move just the mixin to its own file.